core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -67,7 +67,6 @@ class SparkEnv (
     val blockManager: BlockManager,
     val securityManager: SecurityManager,
     val metricsSystem: MetricsSystem,
-    val memoryManager: MemoryManager,
```

Comment:
Could you avoid this breaking change by adding a new constructor, @sunchao ?

Reply:
Sure. I'm pretty much doing something very similar to this PR https://github.com/apache/spark/pull/43627 but happy to add a new constructor.

Reply:
Hmm actually it might be a bit difficult since we are changing `memoryManager` from a `val` to a method.

Reply:
Ack. I also pinged on #43627 . If this is broken already at Spark 4.0.0, we don't need to care much.
- https://github.com/apache/spark/pull/43627#pullrequestreview-1866555431